lib/surface/type_handler.ex

```diff
@@ -204,16 +205,16 @@ defmodule Surface.TypeHandler do
     handler(type).update_prop_expr(value, meta)
   end
 
-  def runtime_prop_value!(module, name, value, node_alias) do
-    type =
+  def runtime_prop_value!(module, name, clauses, opts, node_alias, original, ctx) do
+    {type, _opts} =
       attribute_type_and_opts(module, name, %{
         node_alias: node_alias || module,
-        caller: __ENV__,
-        file: __ENV__.file,
-        line: __ENV__.line
+        caller: %Macro.Env{module: ctx.module},
```

Comment:
is this going to potentially cause issues if we expect caller to be a fully populated env and it only has the module?

Reply:
@lnr0626 I'm not sure what mean. The problem with `runtime_prop_value!` is that since it's called only at runtime, the warnings emitted were all pointing to the the TypeHandler file/module instead of the caller component. The only way I managed to point to the caller was to pass a clean env overriding the module.

Reply:
that makes sense - perhaps a better phrasing of my question is will `%Macro.Env{module: ctx.module}` result in a fully populated `Macro.Env` struct?

Though that's only potentially an issue if we use `ctx.caller` elsewhere in the codebase and expect more than just the `module` attribute to be set

Reply:
oh, I see. Well, the caller context (ctx) was meant to hold as minimum information as possible as it has to be passed as a literal value to `build_assigns`, so we should never have a `ctx.caller`.